src/types.ts

```diff
@@ -0,0 +1,31 @@
+import type { JsonRpcRequest } from 'json-rpc-engine';
+
+export type RequestHeaders = Record<string, string>;
+
+export type ExtendedJsonRpcRequest<T> = JsonRpcRequest<T> & { origin?: string };
+
+/**
+ * These are networks:
+ *
+ * 1. for which Infura has released official, production support (see <https://docs.infura.io>)
+ * 2. which support the JSON-RPC 2.0 protocol
+ */
+export type InfuraJsonRpcSupportedNetwork =
+  | 'mainnet'
+  | 'ropsten'
+  | 'rinkeby'
+  | 'kovan'
+  | 'goerli'
+  | 'eth2-beacon-mainnet'
+  | 'filecoin'
+  | 'polygon-mainnet'
+  | 'polygon-mumbai'
+  | 'palm-mainnet'
+  | 'palm-testnet'
+  | 'optimism-mainnet'
+  | 'optimism-kovan'
+  | 'arbitrum-mainnet'
+  | 'arbitrum-rinkeby'
+  // Legacy networks for compatibility with NetworkController
+  | 'optimism'
```

Comment:
Can we remove these entries in the `NetworkController` instead, since they don't work? Seems wrong to add these here if we know they're invalid.

Reply:
Sure, that's a good point. Removed in [5a4e03f](https://github.com/MetaMask/eth-json-rpc-infura/pull/58/commits/5a4e03f3694e011ec69eaad900d8a04e5b1319b4).